mmpose/core/camera/single_camera_torch.py

```diff
@@ -0,0 +1,118 @@
+        [N, T, J, C]: shape of a batch of pose sequences.
+
+    Args:
+        param (dict): camera parameters including:
+            - R: 3x3, camera rotation matrix (camera-to-world)
+            - T: 3x1, camera translation (camera-to-world)
+            - K: (optional) 2x3, camera intrinsic matrix
+            - k: (optional) nx1, camera radial distortion coefficients
+            - p: (optional) mx1, camera tangential distortion coefficients
+            - f: (optional) 2x1, camera focal length
+            - c: (optional) 2x1, camera center
+        if K is not provided, it will be calculated from f and c.
+
+    Methods:
+        world_to_camera: Project points from world coordinates to camera
+            coordinates
+        camera_to_pixel: Project points from camera coordinates to pixel
+            coordinates
+        world_to_pixel: Project points from world coordinates to pixel
+            coordinates
+    """
+
+    def __init__(self, param, device):
+
+        self.param = {}
+        # extrinsic param
+        R = torch.tensor(param['R'], device=device)
+        T = torch.tensor(param['T'], device=device)
+
+        assert R.shape == (3, 3)
+        assert T.shape == (3, 1)
+        # The camera matrices are transposed in advance because the joint
+        # coordinates are stored as row vectors.
+        self.param['R_c2w'] = R.T
+        self.param['T_c2w'] = T.T
+        self.param['R_w2c'] = R
+        self.param['T_w2c'] = -self.param['T_c2w'] @ self.param['R_w2c']
+
+        # intrinsic param
+        if 'K' in param:
+            K = torch.tensor(param['K'], device=device)
+            assert K.shape == (2, 3)
+            self.param['K'] = K.T
+            self.param['f'] = torch.tensor([[K[0, 0]], [K[1, 1]]],
+                                           device=device)
+            self.param['c'] = torch.tensor([[K[0, 2]], [K[1, 2]]],
+                                           device=device)
+        elif 'f' in param and 'c' in param:
+            f = torch.tensor(param['f'], device=device)
+            c = torch.tensor(param['c'], device=device)
+            assert f.shape == (2, 1)
+            assert c.shape == (2, 1)
+            self.param['K'] = torch.cat([torch.diagflat(f), c], dim=-1).T
+            self.param['f'] = f
+            self.param['c'] = c
+        else:
+            raise ValueError('Camera intrinsic parameters are missing. '
+                             'Either "K" or "f"&"c" should be provided.')
+
+        # distortion param
+        if 'k' in param and 'p' in param:
+            self.undistortion = True
+            self.param['k'] = torch.tensor(param['k'], device=device).view(-1)
+            self.param['p'] = torch.tensor(param['p'], device=device).view(-1)
+            assert len(self.param['k']) in {3, 6}
+            assert len(self.param['p']) == 2
+        else:
+            self.undistortion = False
+
+    def world_to_camera(self, X):
+        assert isinstance(X, torch.Tensor)
```

Comment:
Considering using type hint to replace the assertion.

Reply:
What is type hint?